homeassistant/components/device_tracker/googlehome.py

```diff
@@ -0,0 +1,96 @@
+"""
+Support for Google Home bluetooth tacker.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.googlehome/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.components.device_tracker import (
+    DOMAIN, PLATFORM_SCHEMA, DeviceScanner)
+from homeassistant.const import CONF_HOST
+
+REQUIREMENTS = ['ghlocalapi==0.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_RSSI_THRESHOLD = 'rssi_threshold'
+
+PLATFORM_SCHEMA = vol.All(
+    PLATFORM_SCHEMA.extend({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_RSSI_THRESHOLD, default=-70): vol.Coerce(int),
+    }))
+
+
+async def async_get_scanner(hass, config):
+    """Validate the configuration and return an Google Home scanner."""
+    scanner = GoogleHomeDeviceScanner(hass, config[DOMAIN])
+    await scanner.async_connect()
+    return scanner if scanner.success_init else None
+
+
+class GoogleHomeDeviceScanner(DeviceScanner):
+    """This class queries a Google Home unit."""
+
+    def __init__(self, hass, config):
+        """Initialize the scanner."""
+        from ghlocalapi.device_info import DeviceInfo
+        from ghlocalapi.bluetooth import BluetoothScan
+
+        self.last_results = {}
+
+        self.success_init = False
+        self._host = config[CONF_HOST]
+        self.rssi_threshold = config[CONF_RSSI_THRESHOLD]
+
+        session = async_get_clientsession(hass)
+        self.deviceinfo = DeviceInfo(hass.loop, session, self._host)
+        self.scanner = BluetoothScan(hass.loop, session, self._host)
+
+    async def async_connect(self):
+        """Initialize connection to Google Home."""
+        await self.deviceinfo.get_device_info()
+        data = self.deviceinfo.device_info
+        self.success_init = data is not None
+
+    async def async_scan_devices(self):
+        """Scan for new devices and return a list with found device IDs."""
+        await self.async_update_info()
+        return list(self.last_results.keys())
+
+    async def async_get_device_name(self, device):
+        """Return the name of the given device or None if we don't know."""
+        if device not in self.last_results:
+            return None
+        return self._host + '_' + self.last_results[device]['mac_address']
+
+    async def get_extra_attributes(self, device):
+        """Return the extra attributes of the device."""
+        attributes = {}
+        for attribute in self.last_results[device]:
```

Comment:
You're just making a copy here?

Reply:
Also, are all these attributes of interest ?

Reply:
> You're just making a copy here?

True... I have no idea why I did that...



> Also, are all these attributes of interest ?

This is the current list of attributes:
```text
source_type: bluetooth
latitude: 0
longitude: 0
gps_accuracy: 0
scanner: GoogleHomeDeviceScanner
rssi: -47
btle_mac_address: 43:ef:8b:a9:56:34
ghname: Google Home Living Room
friendly_name: 192.168.2.234 43:ef:8b:a9:56:34
```
`btle_mac_address`, are part of the name, so that can be removed.
`ghname`, and `rssi` are useful if you have multiple devices and are doing room based temating with this information.

Another way, is to allow the definition of multiple IP's have logic to assign the btlemac to the correct base from the rssi of all the colleted units?

example:
```yaml
device_tracker:
  - platform: googlehome
    ip_address:
      - 192.168.1.11
      - 192.168.1.12
```

The platform (library) would then pull info from all defined devices, and return correct values, if a mac address is seen by multiple google home units it will show connected to the one with lowest rssi.

Doing it this way will also look prettier and produce less unessesarry state changes, since a single btlemac would only have one entity no matter how many google home units the user have...

Thinking of it, this makes more sense IMO.

Reply:
Personally I have multiple google homes (1 in each room) and I think most users start with 1 and end up getting more than 1.  Usually one in each room as well.  I think the combined approach would be the best here given most users may have more than 1 device.

Reply:
It doesn't sound bad to combine, but I'd like to reference @balloob's comment here:
https://github.com/home-assistant/architecture/issues/71#issuecomment-421927061

Maybe the suggestion here is not contrary to the architecture comment, but it came to mind anyway.

Reply:
This probably is beyond the scope of this PR, but would it be possible to use triangulate the `rssi` information from multiple GH, if users have, for indoor locations (room level)? I saw a thread not too long back using the library for indoor locations.

Reply:
Martin was right to pull in my comment. We should not combine them.  By combining them, we lose data. For example, we would no longer be able to track movement across the house in the future (based on rssi values).

Reply:
Okei :)
Then I'll consider this conversation as resolved, the only attribute that I see that can be removed is the `btle_mac_address` since it's part of the entity_id but I think it's fine leaving it in there.